mlflow/deployments/cli.py

```diff
@@ -0,0 +1,154 @@
+@parse_custom_arguments
+@deployment_name
+@target_uri
+@click.option("--model-uri", "-m", default=None, metavar="URI",
+              help="URI to the model. A local path, a 'runs:/' URI, or a"
+                   " remote storage URI (e.g., an 's3://' URI). For more information"
+                   " about supported remote URIs for model artifacts, see"
+                   " https://mlflow.org/docs/latest/tracking.html"
+                   "#artifact-stores")  # optional model_uri
+@click.option("--flavor", "-f", help="Which flavor to be deployed. This will be auto "
+                                     "inferred if it's not given")
+def update_deployment(flavor, model_uri, t_uri, name, config):
+    """
+    Update the deployment with ID `deployment_id` in the specified target.
+    You can update the URI of the model and/or the flavor of the deployed model (in which case the
+    model URI must also be specified).
+
+    Additional plugin-specific arguments may also be passed to this command, via `-C key=value`.
+    """
+    config_dict = _user_args_to_dict(config)
+    client = interface.get_deploy_client(t_uri)
+    ret = client.update_deployment(name, model_uri=model_uri, flavor=flavor, config=config_dict)
+    click.echo("Deployment {} is updated (with flavor {})".format(name, ret['flavor']))
+
+
+@commands.command("delete", context_settings=context_settings)
+@deployment_name
+@target_uri
+def delete_deployment(t_uri, name):
+    """
+    Delete the deployment with ID `deployment_id` from the specified target.
+    """
+    client = interface.get_deploy_client(t_uri)
+    client.delete_deployment(name)
+    click.echo("Deployment {} is deleted".format(name))
+
+
+@commands.command("list", context_settings=context_settings)
+@target_uri
+def list_deployment(t_uri):
+    """
+    List the IDs of all model deployments in the specified target. These IDs can be used with
+    the `delete`, `update`, and `get` commands.
+    """
+    client = interface.get_deploy_client(t_uri)
+    ids = client.list_deployments()
+    click.echo("List of all deployments:\n{}".format(ids))
+
+
+@commands.command("get", context_settings=context_settings)
+@deployment_name
+@target_uri
+def get_deployment(t_uri, name):
+    """
+    Print a detailed description of the deployment with ID ``deployment_id`` in the specified
+    target.
+    """
+    client = interface.get_deploy_client(t_uri)
+    desc = client.get_deployment(name)
+    for key, val in desc.items():
+        click.echo("{}: {}".format(key, val))
+    click.echo('\n')
+
+
+@commands.command("help", context_settings=context_settings)
+@target_name
+def target_help(t_name):
+    """
+    Specific help command for deployment plugins. This will call the `target_help` function from
+    the plugin to display the help string specific for each plugin
+    """
+    click.echo(interface.target_help(t_name))
+
+
+@commands.command("run-local", context_settings=context_settings)
```

Comment:
Should update this to have the same arguments as the `create_deployment` CLI command

Reply:
Ok, should we change the command to `deploy-local` or `create-local` then?

Reply:
> Also our model uri follows `:/` as the separator and we are doing `://` as the separator here

This is a good point, using :/ makes more sense to retain parity with other "URI"s in MLflow. I also realize I misread [this URI definition](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Definition) - in general, something like "file:/" or "file:///" is a valid URI, but "file://" in fact is not.

> Ok, should we change the command to deploy-local or create-local then?

`deploy-local` seems reasonable, although I actually slightly prefer `run-local` as it avoids repeating the word "deploy" in the CLI command (e.g. compare `mlflow deployments run-local -t redisai ...` to `mlflow deployments deploy-local -t redisai ...`. I prefer both of these to `create-local` as it suggests that there should also be equivalent CRUD APIs for managing local deployments (e.g. `list-local`, `delete-local`, etc commands)

Reply:
> Also, we don't really need "target URI" for run local, we just need the name of target plugin. Same is applicable in target_help. May be another parameter called --target-name. I don't really like target name on the first look but then advertising --target as target URI wouldn't do a good job here as well

On this point, I think for target_help it'd be good to still accept any valid target URI & show the appropriate help (e.g. if someone tells you to deploy models to `redisai:/prod`, it's nice to be able to use target help on `redisai:/prod` see the help printed vs an error message).

For the `run_local` method, it's a bit trickier, as `mlflow deployments run-local -t redisai:/prod ...` would actually deploy locally vs to your remote target - I think this is also ok though, it's nice to be able to reuse the same target URI across commands/have parallel structure across the APIs, if the user does specify a URI with a path (e.g. `redisai:/prod`) vs just a target name (like `redisai`), we can print a warning to let them know that the model is being deployed locally - no need to add the warning in this PR, we can add it in a follow-up.

Reply:
Awesome. I'll go with `:/` then.
About naming, I prefer `run-local` too, but it doesn't logically make sense because you are not really `running` anything, right? You are deploying it to a local instance and then "run" it later

Reply:
`target_uri` suggestion is sensible. I'll do that

Reply:
Yeah good question - for most deployment targets (sagemaker, kube, azureml, gcp, etc) the user will be running a local server that hosts their model behind a REST API - maybe guess in the redisai case we can make the same argument (that we run a local redisai server for the user if none exists & host the model on it)?

Reply:
Ok, that sounds like a good explanation. Thank you